codalab/worker_manager/slurm_batch_worker_manager.py

```diff
@@ -0,0 +1,295 @@
+        """
+        # Documentation can be found at https://slurm.schedmd.com
+        # Since allocating resource for a worker may take a while, we periodically check
+        # for worker status and remove those workers that failed at starting phase.
+        jobs_to_remove = set()
+        for job_id in self.submitted_jobs:
+            job_acct = self.run_command([self.SACCT, '-j', job_id, '--format', 'state'])
+            if 'FAILED' in job_acct:
+                jobs_to_remove.add(job_id)
+                logger.error("Failed to start job {}".format(job_id))
+            elif 'COMPLETING' in job_acct or 'COMPLETED' in job_acct or 'CANCELLED' in job_acct:
+                jobs_to_remove.add(job_id)
+        self.submitted_jobs = self.submitted_jobs - jobs_to_remove
+        logger.info("Submitted jobs: ".format(self.submitted_jobs))
+
+        # Get all the Slurm workers that are submitted by SlurmWorkerManager and owned by the current user.
+        # Returning result will be in the following format:
+        # JOBID:STATE (header won't be included with "--noheader" option)
+        # 1478828,PENDING
+        # 1478830,PENDING
+        jobs = self.run_command(
+            [self.SQUEUE, '-u', self.username, '--format', '%A,%T', '--noheader']
+        )
+        jobs = jobs.strip().split()
+        logger.info(
+            'Workers: {}'.format(
+                ' '.join(job for job in jobs if job in self.submitted_jobs) or '(none)'
+            )
+        )
+
+        # Get all the RUNNING jobs that are owned by the current user.
+        # Returning result will be in the following format:
+        # JOBID (header won't be included with "--noheader" option)
+        # 1478828
+        # 1478830
+        running_jobs = self.run_command(
+            [self.SQUEUE, '-u', self.username, '-t', 'RUNNING', '--format', '%A', '--noheader']
+        )
+        running_jobs = running_jobs.strip().split()
+
+        return [
+            WorkerJob(active=True) if job in running_jobs else WorkerJob(active=False)
+            for job in self.submitted_jobs
+        ]
+
+    def start_worker_job(self):
+        """
+        Start a CodaLab Slurm worker that submits batch job to Slurm
+        """
+        worker_id = uuid.uuid4().hex
+        # Set up the Slurm worker directory
+        slurm_work_dir = Path(self.args.slurm_work_dir, worker_id)
+        try:
+            slurm_work_dir.mkdir(parents=True, exist_ok=True)
+        except PermissionError as e:
+            logger.error(
+                "Failed to create the Slurm work directory: {}. "
+                "Stop creating a new worker.".format(e)
+            )
+            return
+
+        # This needs to be a unique directory since Batch jobs may share a host
+        worker_network_prefix = 'cl_worker_{}_network'.format(worker_id)
+        # Codalab worker's work directory
+        worker_work_dir = Path(
+            self.args.worker_work_dir_prefix, 'codalab-worker-scratch', worker_id
```

Comment:
Can you change `codalab-worker-scratch` to `codalab-slurm-worker-scratch`. Also, instead of just `worker_id`, can you do `username + worker_id` so it's easier to quickly figure out which ones belong to a particular user?